src/dpl/src/dbToOpendp.cpp

```diff
@@ -261,7 +262,16 @@ void Opendp::makeGroups()
           box = box.intersect(core_);
           // offset region to core origin
           box.moveDelta(-core_.xMin(), -core_.yMin());
-
+          if (height == *(unique_heights.begin())) {
+            bgBox bbox(
+                bgPoint(box.xMin(), box.yMin()),
+                bgPoint(
+                    box.xMax() - 1,
+                    box.yMax()
+                        - 1));  /// the -1 is to prevent imagianring overlaps
```

Comment:
imagianring  ?

An alternative is to check that the overlap area is non-zero.  Does shrinking the rect imply an issue if an instances touches the region edge?

Reply:
I noticed this issue when I was debugging the failing fence01.tcl test
The regions are defined in the file as follows:
```
REGIONS 2 ;
- er0 ( 0 0 ) ( 67780 30800 ) + TYPE FENCE ;
- er1 ( 0 30800 ) ( 67780 61600 ) + TYPE FENCE ;
```

The point 0, 30800 and all the points on that horizontal line seems to be inside both regions, which I think is meaningless. Thus, I assumed the first region would start at 0 and end at (30800 - 1)  and the new one starts at 30800. The same applies to the vertical boundaries in other test cases. 

Without this, if the instance touched 30800, the r-tree found 2 overlaps, one with each region. I think my code doesn't change anything but the r-tree regions. The cell regions are untouched, and it would still be from 0 to 30800.

Reply:
tl; dr;
This is not affecting the cell regions themselves, but rather the r-tree regions that is used to count the overlaps. I assume 0 to 30800 is exclusive of the right-boundary -> [0, 30800) and [30800, 61600)

This is because the chip size in the y-dimensions is 61600, yet from 0 to 61600 is 61601. So my assumption makes sense. Otherwise, the regions defined in all the DEF files need modification to avoid overlaps.

Reply:
imagianring is a typo. I meant imaginary.

Reply:
If you are insetting only the top/right but not the bottom/left wouldn't you have a similar issue on those edges?

Reply:
I am a bit confused. Can you elaborate what is the problem you are referring to? 
I am inserting two points that define the bottom-left and the top-right corners. They are used to define the rectangle between them that is inserted in the r-tree.

Reply:
You subtract one from the max points but you don't add one to the min points.

Reply:
yes, but for example if region 1 is from 0 to 5 and the other region is from 5 to 10. Then region 1 includes the 0, but not 5. similarly, region 2 would include 5 and not the 10.

I don't think I need to add 1 to the min as subtracting 1 from the max assures no overlaps.